### DISCUSSION TITLE
Complete the task based o the process response.

### DISCUSSION THREAD

**Discussion by chethangowda** [kagkarlsson/db-scheduler#820] [General]
My code is here:

In Configuration

    @Bean
    public static Task<Void> foo() {
    return Tasks.oneTime("foo")
        .execute((instance, ctx) -> {
            System.out.printf("FOO One Shot Task: %s, ctx: %s\n", instance, ctx);
            callingSomeMethod()
        });
    }

Scheduling the task here

`scheduler.schedule(SchedulerBean.foo().instance("1045"), Instant.now().plusSeconds(10));`

My requirement is:
when task is received, we call the method **callingSomeMethod()** 

1. The task should not be removed from the table until the method returns a success response.
2. The task should be restarted if method gives failure | something fails | returns false

Basically, can we delete or reschedule the task manually?

Thanks for any help!

**Comment by kagkarlsson** [comment]
This is possible. Have a look at `CustomTask`, for example see [LongRunningJobConfiguration.java](https://github.com/kagkarlsson/db-scheduler/blob/master/examples/spring-boot-example/src/main/java/com/github/kagkarlsson/examples/boot/config/LongRunningJobConfiguration.java)
